engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -241,7 +242,7 @@ public LogChannelInterface getLogChannel() {
     return log;
   }
 
-  public String getXML() {
+  public synchronized String getXML() {
```

Comment:
Sorry, Marc, you are wrong here.
Lets assume we have a a slaveServer shared among 2 threads (Thread-1 and Thread-2).
Lets assume getXml() method is not synchronized.

Thread-1 calls getXml() method and in the middle of XML-method execution Thread-2 calls replaceMeta method. replaceMeta begins mutating fields of sharedSlaveServer object.
As a result getXml() method can mix up it's variables (one half of variables can be from the set before replaceMeta call here, another half - after replaceMeta call).

Synchronization is needed here, agree?